cs-front/components/custom/SignInForm.tsx

```diff
@@ -1,11 +1,65 @@
-import React from "react";
+'use client'; // Add this line at the top
+import React, { useState } from "react";
 import Link from "next/link";
 import Image from "next/image";
 import { Label } from "@/components/ui/label";
 import { Input } from "@/components/ui/input";
 import { Button } from "@/components/ui/button";
+import { useRouter } from "next/router";
+//import { useRouter } from 'next/router'; // Import useRouter
 
 export default function SignInForm() {
+  // const router = useRouter(); // Initialize router
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [successMessage, setSuccessMessage] = useState('');
+
+  const handleLogin = async (e: { preventDefault: () => void; }) => {
+    e.preventDefault(); // Prevent default form submission
+
+    const credentials = btoa(`${email}:${password}`); // Base64 encode credentials
+    console.log("email " + email);
+    console.log("passwiord " + password);
+    try {
+
+      const myHeaders = new Headers();
+      myHeaders.append("Content-Type", "application/json");
+      // Comment out the Authorization header if not needed
+      // myHeaders.append("Authorization", `Basic ${credentials}`);
+
+      const raw = JSON.stringify({
+        email, // shorthand for { email: email }
+        password,
+      });
+
+      const requestOptions = {
```

Comment:
Doesn't matter too much for this project since privacy isn't a huge concern rn but I would add the hashing and salting logic to the frontend here and then send that as the raw rather than doing it on the backend just for security reasons in case someone is listening in the middle.